src/components/layout/Layout.tsx

```diff
@@ -124,12 +148,68 @@ const Okp4SocialMedia = ({
   )
 }
 
+// eslint-disable-next-line max-lines-per-function
 const Layout = ({ config, children }: DeepReadonly<LayoutProps>): JSX.Element => {
   const { theme }: ThemeContextType = useTheme()
   const { t }: UseTranslationResponse = useTranslation()
   const isLightTheme = theme === 'light'
   const themedImage = isLightTheme ? lightCosmos.src : darkCosmos.src
   const footerLabel = t('footer:brand-link')
+  const {
+    createUrl,
+    exploreUrl,
+    interactUrl,
+    learnUrl,
+    okp4Url
+  }: NavigationMenuUrls | Record<string, undefined> = config?.app.navigationMenuUrls ?? {
+    createUrl: undefined,
+    exploreUrl: undefined,
+    interactUrl: undefined,
+    learnUrl: undefined,
+    okp4Url: undefined
+  }
+
+  const menuItems: MenuItem[] = [
+    {
+      namespace: 'header:create',
+      url: createUrl,
+      externalLink: true
+    },
+    {
+      namespace: 'header:explore',
+      url: exploreUrl,
+      externalLink: false
+    },
+    {
+      namespace: 'header:interact',
+      url: interactUrl,
+      externalLink: true
+    },
+    {
+      namespace: 'header:learn',
+      url: learnUrl,
+      externalLink: true
+    },
+    {
+      namespace: 'header:okp4',
+      url: okp4Url,
+      externalLink: true
+    }
+  ]
+
+  const navigationItems: NavigationItem[] = menuItems.map(
+    ({ url, namespace, externalLink }: DeepReadonly<MenuItem>) => ({
+      menuItem: (
+        <Typography as="div" fontSize="small" fontWeight="bold" key={uuidv4()} noWrap>
```

Comment:
Why do you use a `uuid` as key?

Reply:
Oh! 🧐 Well spotted. Keys must be *stable*. Don't use uuid for that as it assigns new key for each element every time... Since your list is constant, just use the index, it will do the trick.

Reply:
It was me who added this feature and I didn't want to use the index as it's not recommended, so I choose to use uuid to generate a unique key.

Reply:
Random keys are worst than indexes about performance and consistency.
Indexes are deprecated because they are weak (element could be moved or removed).
Keys must be under control, so please remove random uuids and use values instead (url or concatenation of values).

Reply:
Oki I was not aware of this, thanks for explaining 🙂  I will use the `namespace` instead as the url could be undefined which if I'm not mistaken would cause react to default to using index.

Reply:
@ErikssonJoakim 
From the React docs:

```
Keys should be stable, predictable, and unique. Unstable keys (like those produced by Math.random()) will cause many component instances and DOM nodes to be unnecessarily recreated, which can cause performance degradation and lost state in child components.
```

Reply:
> Oki I was not aware of this, thanks for explaining 🙂  I will use the `namespace` instead as the url could be undefined which if I'm not mistaken would cause react to default to using index.

I'd just use the index since the list is constant.